providers/infrastructure-vsphere/v1.4.1/cconly/base.yaml

```diff
@@ -930,6 +990,19 @@ spec:
                   ,
                 {{- end }}
                 {{- .key -}} = {{- .value -}}
+                {{- $nodePoolLabelNotExist = false }}
+              {{- end }}
+            {{- end }}
+            {{- if $nodePoolLabelNotExist | and .worker.nodeLabels -}}
+              ,
+              {{- $first := true }}
+              {{- range .worker.nodeLabels }}
+                {{- if $first }}
+                  {{- $first = false }}
+                {{- else -}}
+                  ,
+                {{- end }}
+                {{- .key -}} = {{- .value -}}
```

Comment:
@tenczar Thank for taking a look, Timmy is the PM I was confirming with, if just for tkgm, this implementation also break `tanzu node-pool` cli? I don't understand how.  If you mean in general (regardless of tkgm or tkgs), this breaks `tanzu node-pool` since tanzu cli is for both tkgm and tkgs, then I agree.
I agree with you that this introduces 2 sets of logic for tkgm and tkgs, I will confirm whether Timmy also owns tkgs.

Reply:
Could you help explain why this implementation break `tanzu node-pool` for tkgm? Any additional fix needed if just for tkgm works?

Reply:
@tenczar Does "throwing out nodePoolLabels and just using worker.nodeLabels" still break the tanzu node-pool cli functionality?